libpod/container_exec.go

```diff
@@ -1306,3 +1306,60 @@ func justWriteExecExitCode(c *Container, sessionID string, exitCode int, emitEve
 	// Finally, save our changes.
 	return c.save()
 }
+
+// ExecNoSession is a high-level wrapper for execNoSession.
+func (c *Container) ExecNoSession(config *ExecConfig, streams *define.AttachStreams, resize <-chan resize.TerminalSize) (int, error) {
+	return c.execNoSession(config, streams, resize, false)
+}
+
+// execNoSession executes a command in a container without creating a persistent exec session.
+// It skips database operations and minimizes container locking for performance.
+func (c *Container) execNoSession(config *ExecConfig, streams *define.AttachStreams, _ <-chan resize.TerminalSize, _ bool) (exitCode int, retErr error) {
+	sessionID := stringid.GenerateRandomID()
+
+	session := &ExecSession{
+		Id:     sessionID,
+		Config: config,
+	}
+
+	if !c.batched {
```

Comment:
This should move up into `ExecNoSession` - the lowercase, internal function should be unlocked.